packages/realm/binding/android/build.gradle

```diff
@@ -59,7 +59,7 @@ android {
                           "-DCMAKE_CXX_VISIBILITY_PRESET=hidden",
                           "-DREACT_NATIVE_ROOT_DIR=${REACT_NATIVE_ROOT_DIR}"
                 targets 'realm-js-android-binding'
-                cppFlags '-std=c++20'
+                cppFlags '-O2 -std=c++20 -frtti -fexceptions -Wall -fstack-protector-all'
```

Comment:
Should we use `-Os` or `-Oz` instead?

Reply:
It doesn't seem to make too much of a difference:

### `-O2 -std=c++20 -frtti -fexceptions -Wall -fstack-protector-all`

```
❯ du -h ../../../packages/realm/binding/android/src/main/jniLibs/*/librealm.so
 12M    ../../../packages/realm/binding/android/src/main/jniLibs/arm64-v8a/librealm.so
6.9M    ../../../packages/realm/binding/android/src/main/jniLibs/armeabi-v7a/librealm.so
 12M    ../../../packages/realm/binding/android/src/main/jniLibs/x86/librealm.so
 13M    ../../../packages/realm/binding/android/src/main/jniLibs/x86_64/librealm.so
 ```
 
 ### `-Os -std=c++20 -frtti -fexceptions -Wall`

```
❯ du -h ../../../packages/realm/binding/android/src/main/jniLibs/*/librealm.so
 12M    ../../../packages/realm/binding/android/src/main/jniLibs/arm64-v8a/librealm.so
6.8M    ../../../packages/realm/binding/android/src/main/jniLibs/armeabi-v7a/librealm.so
 12M    ../../../packages/realm/binding/android/src/main/jniLibs/x86/librealm.so
 13M    ../../../packages/realm/binding/android/src/main/jniLibs/x86_64/librealm.so
 ```
 
 ### `-Oz -std=c++20 -frtti -fexceptions -Wall`

```
❯ du -h ../../../packages/realm/binding/android/src/main/jniLibs/*/librealm.so
 12M    ../../../packages/realm/binding/android/src/main/jniLibs/arm64-v8a/librealm.so
6.8M    ../../../packages/realm/binding/android/src/main/jniLibs/armeabi-v7a/librealm.so
 12M    ../../../packages/realm/binding/android/src/main/jniLibs/x86/librealm.so
 13M    ../../../packages/realm/binding/android/src/main/jniLibs/x86_64/librealm.so
 ```

Reply:
We could probably leave out the `stack-protector-all` flag, as it does come with a runtime cost and we're taking user input data through JSI values which I think does provide some protection to these type of issues. I'd rather that we opt on the safe side and do some investigations to see if we could reasonably do without it, if we run a project to squeeze out some performance later on.